### ISSUE TITLE
Refactor seedlots to better define relationships between source and content

### ISSUE THREAD

**Issue by bellerbrock** [solgenomics/sgn#2740]
Expected Behavior 
--------------------------------------------------------------------------
There are a lot of issues related to better tracking the flow of breeding material through a breeding program (#1817, #236, #144)

Seedlots are the best way to track the movement of seeds from a cross to a plot in year one, then from the year one plot to more plots in year two, etc. The problem is that there is not a clear distinction between source (a specific plot, or specific cross), and contents (a specific genotype, or a specific family/population). And only accessions OR crosses are currently allowed as content, there is no way to relate to more categories or both categories.

This causes problems for clonal crops, because the key relationship year to year is the seedlot from one season's plot providing planting material for next season's plot. 

It also causes problems for cross-breeding crops because seedlots from a selfed inbred, or seedlots from a a pair of inbreds crossed to form a hybrid, are genetically uniform and synonymous with accession. In the current system if you add such a seedlot with the cross as contents it becomes harder to use it as the source of seed for a specific accession in a trial. If you add it with the accession as contents instead then you lose track of the exact nursery and plot or plant it came from.

So long story short, the idea is to distinguish between source and contents, allowing plant, plot, cross, or (in cases where neither is known) accession as source, and accession or family/population as contents.

For Bugs:
---------

**Comment by titima15** [comment]
@bellerbrock Because specific plot or plant can be recorded as parents for cross, it means seedlots which are linked to cross will automatically have information about plot or plant.

**Comment by bellerbrock** [comment]
@titima true, that's a key benefit of creating seedlots from crosses. The problem though is that the cross is really the source, rather than the content, of the seedlot. Depending on the situation the contents of a seedlot that come from a cross are either a segregating population/family, or a genetically uniform inbred or hybrid (aka an accession).
Without tracking the source and contents as distinct properties it becomes impossible to fully represent to flow of genetic material through a breeding program.
Let me know if that makes sense, if not maybe I can dig up some examples from sweetpotato or maize.

**Comment by titima15** [comment]
@bellerbrock cross type can imply the flow of genetic material as well.

**Comment by titima15** [comment]
In fact, cross type and information about parents are the keys for indicating the flow of genetic material.

**Comment by bellerbrock** [comment]
@titima that's a good point. In an ideal system you could infer the contents of the seedlot that comes from a cross based on the cross type and the properties of the parents involved in the cross. A biparental cross between two inbreds would always produce the same hybrid, a self of an inbred would always produce the same inbred. A biparental cross between two hybrids would always produce the same segregating F1 population, etc.
In reality though you won't always know all of the source details, or even if you do they may not be tracked in the database. So for now at least it will be important to track both the source and the contents. The source property can be optional though, to deal with cases where it's unknown.

**Comment by bauchetg** [comment]
Great discussion @bellerbrock, a quick summary:
Break seedlot into two types" "botanical" vs "clonal" and subsequent implications "botanica"l as source (ie: cross) but not necessarily content (ie plot) where "clonal" as content but necessarily source. Exceptions may apply.

**Comment by lukasmueller** [comment]
I think I added a feature similar to this. In the upload file, an "alternate source" can be added that will be used in the first transaction, for example, a plot that is the provenance of the seed. This is used a lot for seed multiplication experiments, which usually have not separate crossing experiment information.
